### ISSUE TITLE
Error in submission: sha1 file '<stdout>' write error: Broken pipe

### ISSUE THREAD

**Issue by xlouc** [go-gitea/gitea#7547]
- Gitea version (or commit ref): 1.9.0-rc2
- Git version: 1.8.3.1
- Operating system: CentOS Linux 7 (Core)
- Database (use `[x]`):
  - [ ] MSSQL
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
SSH submitted the code with the following error (not using HTTPS links, but using git@git.xx.xx)
```
fatal: protocol error: expected old/new/ref, got 'shallow 3b74753d7c38f7a9d3e98b2314b172724bdc8a7c'
Gitea: Internal error
Failed to execute git command: exit status 128
fatal: sha1 file '<stdout>' write error: Broken pipe
fatal: the remote end hung up unexpectedly
fatal: the remote end hung up unexpectedly
```
The git client is also configured
```
[http]
postBuffer = 524288000
```

**Comment by lunny** [comment]
Please speak English.

**Comment by mamying** [comment]
@lunny Has been translated into English

**Comment by lafriks** [comment]
Check gitea.log for errors

**Comment by mamying** [comment]
no errors

**Comment by lunny** [comment]
@mamying  Maybe because your git version is too low.

**Comment by lunny** [comment]
This maybe fixed on v1.10.1 could you try that?
